MekHQ/src/mekhq/campaign/parts/MissingPart.java

```diff
@@ -419,18 +421,19 @@ public void reservePart() {
 
     @Override
     public void cancelReservation() {
-        Part replacement = findReplacement(false);
-        if (hasReplacementPart() && (null != replacement)) {
-            setReplacementPart(null);
-            replacement.setReservedBy(null);
-            if (replacement.isSpare()) {
-                Part spare = campaign.getWarehouse().checkForExistingSparePart(replacement);
-                if (null != spare) {
-                    spare.incrementQuantity();
-                    campaign.getWarehouse().removePart(replacement);
+        if (hasReplacementPart()) {
+            Part replacement = getReplacementPart();
+            if (replacement != null) {
+                replacement.setReservedBy(null);
+
+                // Only return the replacement part to the campaign if we have one
+                if (replacement.getQuantity() > 0) {
+                    campaign.getQuartermaster().addPart(replacement, 0);
```

Comment:
Remind me, what's the second argument here?

Reply:
`transitDays`, should probably name the method better and add an immediate version.

Something like:
```java
quartermaster.addInTransitPart(part, transitDays);
quartermaster.addPart(part);
```